internal/cortex/frontend/transport/handler.go

```diff
@@ -122,14 +171,88 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	r.Body = http.MaxBytesReader(w, r.Body, f.cfg.MaxBodySize)
 	r.Body = io.NopCloser(io.TeeReader(r.Body, &buf))
 
+	// Check if caching is enabled
+	if f.lru != nil {
+		// Store query expression
+		queryExpressionNormalized = f.regex.ReplaceAllString(r.URL.Query().Get("query"), " ")
+
+		// Time range length for queries, if either of "start" or "end" are not present, return 0
+		getQueryRangeSeconds := func() int {
+			start, err := strconv.Atoi(r.URL.Query().Get("start"))
+			if err != nil {
+				return 0
+			}
+			end, err := strconv.Atoi(r.URL.Query().Get("end"))
+			if err != nil {
+				return 0
+			}
+			return end - start
+		}
+
+		//Store query time range length
+		queryExpressionRangeLength = getQueryRangeSeconds()
+
+		// Check if query in cache and whether value exceeds time range length
+		if value, ok := f.lru.Get(queryExpressionNormalized); ok && value.(int) >= queryExpressionRangeLength {
+			w.WriteHeader(http.StatusForbidden)
+			level.Warn(util_log.WithContext(r.Context(), f.log)).Log("msg", "FOUND QUERY IN CACHE: CAUSED ERROR: ", "query expression", queryExpressionNormalized)
+			//f.cachedHits.WithLabelValues(userID).Add(1)
```

Comment:
Do this. If `userID` is not in this context, remove this label in the counter registration. Use `[]string{}` instead of `[]string{"user"}`.

```
f.cachedHits.WithLabelValues(userID).Add(float64(1))
```